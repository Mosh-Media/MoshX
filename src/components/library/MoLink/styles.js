const styles = (theme) => ({
  link: {
    fontFamily: theme.fontFamily,
    textDecoration: "none",
    padding: "0",
    color: theme.blue?.medium,
    fontSize: "1.1rem",
    borderRadius: "3px",
    textTransform: "initial",
    fontWeight: "600",
    "&:hover .arrow": {
      transform: "translateX(5px)",
    },
  },
  icon: {
    transition: "transform .35s",
    paddingLeft: "4px",
    paddingTop: "2px",
  },
});

export default styles;
