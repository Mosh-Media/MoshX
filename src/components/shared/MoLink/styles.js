const styles = theme => ({
  link: {
    background: "none",
    textDecoration: "none",
    padding: "0px",
    color: theme.blue?.medium,
    fontSize: "1.1rem",
    borderRadius: "3px",
    textTransform: "initial",
    fontWeight: "600",
    "&:hover": {
      background: "none"
    },
    "&:hover arrow": {
      transform: "translateX(5px)"
    }
  },
  arrow: {
    fontSize: "1.0rem",
    paddingLeft: "4px"
  }
});

export default styles;
