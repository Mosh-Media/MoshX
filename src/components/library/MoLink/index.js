/**
 * MoLink
 * Browser and Editor mocks are styled to look like a browser to frame the code editor and preview panels
 *
 * @param {String} href - required route to navigate to, otherwise it will be a # for route
 * @param {String} text - text displayed for the button
 */

import React from 'react';

import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
import { Button } from '@material-ui/core';
import { Link } from 'react-router-dom';
import withStyles from '@material-ui/core/styles/withStyles';
import styles from './styles';

const MoLink = ({ classes, href, text }) => {
  return (
    <Button component={Link} to={href || '#'} className={classes.link}>
      {text}
      <ArrowForwardIcon className={`${classes.icon} arrow`} />
    </Button>
  );
};

export default withStyles(styles)(MoLink);
