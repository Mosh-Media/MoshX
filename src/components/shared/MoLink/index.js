import React from "react";

import { Link } from "react-router-dom";
import styles from "./styles";
import withStyles from "@material-ui/core/styles/withStyles";

const MoButton = ({ classes, href, text }) => {
  return (
    <>
      <Link to={href ? href : "#"} className={classes.link}>
        {text}
      </Link>
    </>
  );
};

export default withStyles(styles)(MoButton);
