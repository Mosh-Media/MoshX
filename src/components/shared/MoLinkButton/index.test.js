import React from "react";
import MoLink from "./index.js.js";
import renderer from "react-test-renderer";

it("Should render MoLink", () => {
  const tree = renderer.create(<MoLink />).toJSON();
  expect(tree).toMatchSnapshot();
});
