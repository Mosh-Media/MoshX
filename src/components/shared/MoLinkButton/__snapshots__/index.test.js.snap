// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MoLink 1`] = `
<a
  aria-disabled={false}
  className="MuiButtonBase-root MuiButton-root MuiButton-text MoLinkButton-link-1"
  href="/"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="MuiButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root arrow"
      focusable="false"
      role="presentation"
      style={
        Object {
          "fontSize": "1em",
          "paddingLeft": "4px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
  </span>
</a>
`;
